posix.mak

```diff
@@ -375,6 +377,16 @@ unittest/%.run : $(ROOT)/unittest/test_runner
 %.test : $(LIB)
 	$(MAKE) -f $(MAKEFILE) $(addsuffix .test,$(patsubst %.d,%,$(wildcard $*/*)))
 
+%.debug_with_debugger : %.d $(LIB)
+	$(DMD) $(DFLAGS) -main -unittest $(LIB) -defaultlib= -debuglib= $(LINKDL) $<
```

Comment:
Generally our makefiles (and makefiles are general) are full of hacks that might be difficult to wrap your head around, especially for people new to make. Thus more comments for what things do and why they things are being done that way are never unwelcome.

In this case, I think some things that wouldn't hurt to document via inline comments are:
- Explain what the `.debug_with_debugger` is for and why it's needed
- Explain why a recursive make invocation is necessary
- Explain why this compiles a single file and then links it against the Phobos library (the answer to which is my comment above)

Please also improve the commit message. See https://chris.beams.io/posts/git-commit/, especially point 7.

These are not blockers but one of the avenues of improvement we've talked about is to improve commit messages and technical documentation in complicated parts of the codebase such as the compiler and makefiles, to make it easier for everyone to contribute as well as review past and proposed changes.